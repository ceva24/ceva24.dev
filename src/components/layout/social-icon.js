import React from "react"
import { css } from "@emotion/core"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { rhythm } from "src/utils/typography"

export default (props) => (
    <a href={props.url} title={props.url} aria-label={props.label} css={css`
        color: inherit;
        margin-right: ${rhythm(1 / 2)};
    `}>
        <FontAwesomeIcon icon={props.icon} />
    </a>
)
