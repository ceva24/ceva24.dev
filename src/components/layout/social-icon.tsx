import { css } from "@emotion/react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { IconProp } from "@fortawesome/fontawesome-svg-core";
import { rhythm } from "../../utils/typography";

type SocialIconProps = {
    url: string;
    label: string;
    icon: IconProp;
};

const SocialIcon = ({ url, label, icon }: SocialIconProps) => (
    <a
        href={url}
        title={url}
        aria-label={label}
        css={css`
            color: inherit;
            margin-right: ${rhythm(1 / 2)};
        `}
    >
        <FontAwesomeIcon icon={icon} />
    </a>
);

export {SocialIcon };
