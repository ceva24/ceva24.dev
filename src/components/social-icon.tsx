import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import type { IconProp } from "@fortawesome/fontawesome-svg-core";

interface SocialIconProps {
    url: string;
    label: string;
    icon: IconProp;
}

const SocialIcon: React.FC<SocialIconProps> = ({
    url,
    label,
    icon,
}: SocialIconProps) => (
    <a href={url} title={url} aria-label={label} className="text-gray-800">
        <FontAwesomeIcon icon={icon} />
    </a>
);

export { SocialIcon };
