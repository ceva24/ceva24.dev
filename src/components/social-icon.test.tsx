import React from "react";
import { render } from "@testing-library/react";
import { faTwitch } from "@fortawesome/free-brands-svg-icons";
import { SocialIcon } from "./social-icon";

describe("social icon", () => {
    it("renders", () => {
        const { asFragment } = render(
            <SocialIcon
                url="https://twitch.tv"
                label="Twitch"
                icon={faTwitch}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
