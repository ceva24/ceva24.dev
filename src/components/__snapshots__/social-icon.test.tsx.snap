// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social icon renders 1`] = `
<DocumentFragment>
  <a
    aria-label="Twitch"
    class="text-gray-800"
    href="https://twitch.tv"
    title="https://twitch.tv"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-twitch "
      data-icon="twitch"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M391.17,103.47H352.54v109.7h38.63ZM285,103H246.37V212.75H285ZM120.83,0,24.31,91.42V420.58H140.14V512l96.53-91.42h77.25L487.69,256V0ZM449.07,237.75l-77.22,73.12H294.61l-67.6,64v-64H140.14V36.58H449.07Z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;
